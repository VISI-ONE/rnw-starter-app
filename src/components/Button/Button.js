import React from 'react';
import { string, func, number, object, oneOfType } from 'prop-types';
import { Text, TouchableHighlight } from 'react-native';

const Button = ({
  style,
  onClick,
  text,
  textStyle
}) => (
  <TouchableHighlight onPress={onClick} style={style} >
    <Text style={textStyle}>{text}</Text>
  </TouchableHighlight>
);

Button.propTypes = {
  style: oneOfType([object, string, number]),
  textStyle: oneOfType([object, string, number]),
  onClick: func,
  text: string
};

export default Button;
