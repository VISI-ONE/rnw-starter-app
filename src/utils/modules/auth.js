import { AsyncStorage } from 'react-native';
import decodeJwt from './decodeJwt';
import { isDev } from '../common';
const TOKEN_KEY = 'authtoken';

const request = (url, method, body, extraHeaders) => {
	let headers = {
		'Content-Type': 'application/json'
	};

	if (extraHeaders) {
		headers = { ...headers, ...extraHeaders };
	}

	const options = {
    method,
    headers,
  };

	if (method === 'POST') {
		options.body = body;
	}
	return fetch(url, options)
  .then(res => res.json());
};

export const logout = () => {
	AsyncStorage.removeItem(TOKEN_KEY);
};

export const login = (username, password) => {
  return request('some-url', 'POST', JSON.stringify({ username, password }))
	.then(res => {
		if (res && res.status === 'success' && res.token) {
			setUserToken(res.token);
		}
		return res;
	})
  .catch(err => err);
};

export const isTokenExpired = (token) => {
	try {
		const decodedToken = decodeJwt(token);
		const tokenExperationTimestamp = parseInt(decodedToken.exp, 10); // eg 1505727610 // seconds since UTC 1970.. .

		const dateNow = new Date();
		const clientTimestamp = Math.round(dateNow.getTime() / 1000);

		if (tokenExperationTimestamp < clientTimestamp) {
			// TokenExperation is in the past (less then Now)
			return true;
		}
		// TokenExperation is in the future (more then Now)
		return false;
	} catch (e) {
		return true;
	}
};

export const getUserToken = () => {
	return AsyncStorage.getItem(TOKEN_KEY)
	.then(token => {
		if (token && !isTokenExpired(token)) {
			return token;
		}
		return undefined;
	});
};

export const setUserToken = (token) => {
	if (token) {
		try {
			AsyncStorage.setItem(TOKEN_KEY, token);
		} catch (e) {} // eslint-disable-line no-empty
	}
};

export const checkUserToken = () => {
	return getUserToken()
	.then(token => {
		if (token) {
			return request('refresh url', 'GET', null, { Bearer: token })
			.then(res => {
				if (res && res.status === 'success' && res.newToken) {
					setUserToken(res.newToken);
				}
				return res;
			})
		  .catch(err => err);
		}
		return null;
	});
};
