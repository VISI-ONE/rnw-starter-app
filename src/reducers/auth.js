import { LOGIN, LOGOUT } from '../actions/types';
import { logout } from '../utils/modules/auth';

const initialState = {
  user: null,
  isLoggedIn: false
};

export default (state = initialState, action) => {
  switch (action.type) {
    case LOGIN:
      return {
        ...state,
        user: action.user,
        isLoggedIn: true
      };
    case LOGOUT:
      logout();
      return {
        ...state,
        user: null,
        isLoggedIn: false
      };
    default:
      return state;
  }
};
