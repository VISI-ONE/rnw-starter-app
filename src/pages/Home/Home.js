import React, { Component } from 'react';
import { connect } from 'react-redux';
import { StyleSheet, View, Text, Image } from 'react-native';
import { logout as logoutAction } from '../../actions/auth';
import Button from '../../components/Button';

const logo = require('../../assets/logo.png');

export class Home extends Component {
  logout = () => {
    const { logout } = this.props;
    logout();
  }

  render() {
    const { translations } = this.props;
    return (
      <View style={styles.container}>
        <View style={styles.content}>
          <Text style={styles.biggerText}>{translations.home__welcome}</Text>
          <Image
            source={logo}
            style={styles.image}
          />
          <View style={styles.row}>
            <Button style={styles.button} text={translations.logout} onClick={this.logout} textStyle={styles.buttonText} />
          </View>
          <Text style={styles.text}>
            {translations.home__loggedin}
          </Text>
        </View>
      </View>
    );
  }
}

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignSelf: 'stretch',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    paddingTop: 10,
  },
  content: {
    flex: 1,
  },
  row: {
    flexDirection: 'row',
    margin: 5,
    justifyContent: 'center',
    alignItems: 'center'
  },
  biggerText: {
    fontSize: 17,
    alignSelf: 'center'
  },
  button: {
    margin: 5
  },
  buttonText: {
    color: '#00F',
  },
  image: {
    width: 300,
    height: 300
  }
});

export const mapStateToProps = state => ({
  translations: {
    home__loggedin: state.locale.translations.home__loggedin,
    home__welcome: state.locale.translations.home__welcome,
    logout: state.locale.translations.logout,
  }
});

export const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(logoutAction())
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);
