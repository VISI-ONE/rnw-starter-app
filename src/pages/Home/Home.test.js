import { View, Text, Image, Platform } from 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { styles, mapStateToProps, mapDispatchToProps, Home } from './Home';
import { logout as logoutAction } from '../../actions/auth';
import Button from '../../components/Button/Button';
import translations from '../../utils/locales/en-US';

describe('Home page', () => {
  describe('rendering', () => {
    beforeEach(() => {
      fetch.resetMocks();
    });
    it('render properly - ' + Platform.OS, () => {
      fetch.mockResponseOnce(JSON.stringify({}));
      const login = jest.fn();
      const logout = jest.fn();
      const tree = renderer.create(<Home login={login} logout={logout} translations={translations} />);
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('render properly with qr code - ' + Platform.OS, () => {
      fetch.mockResponseOnce(JSON.stringify({}));
      const logout = jest.fn();
      const tree = renderer.create(
        <Home logout={logout} translations={translations} />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('render the proper structure', () => {
      fetch.mockResponseOnce(JSON.stringify({}));
      const logout = jest.fn();
      const tree = shallow(<Home logout={logout} translations={translations} />);
      const content = shallow(
        <View style={styles.container}>
          <View style={styles.content}>
            <Text style={styles.biggerText}>{translations.home__welcome}</Text>
            <Image
              source={require('../../assets/logo.png')}
              style={styles.image}
            />
            <View style={styles.row}>
              <Button style={styles.button} text="Logout" onClick={logout} textStyle={styles.buttonText} />
            </View>
            <Text style={styles.text}>{translations.home__loggedin}</Text>
          </View>
        </View>
      );

      expect(tree.html()).toEqual(content.html());
    });

    it('render the proper structure with qr code', () => {
      const logout = jest.fn();
      const tree = shallow(<Home logout={logout} translations={translations} />);
      const content = shallow(
        <View style={styles.container}>
          <View style={styles.content}>
            <Text style={styles.biggerText}>{translations.home__welcome}</Text>
            <Image
              source={require('../../assets/logo.png')}
              style={styles.image}
            />
            <View style={styles.row}>
              <Button style={styles.button} text="Logout" onClick={logout} textStyle={styles.buttonText} />
            </View>
            <Text style={styles.text}>{translations.home__loggedin}</Text>
          </View>
        </View>
      );

      expect(tree.html()).toEqual(content.html());
    });
  });

  describe('functionality', () => {
    beforeEach(() => {
      fetch.resetMocks();
    });
    it('trigger logout onClick calls logout action', () => {
      fetch.mockResponseOnce(JSON.stringify({}));
      const logout = jest.fn();
      const tree = shallow(<Home logout={logout} translations={translations} />);
      const btn = tree.find({ text: 'Logout' });

      btn.simulate('click');
      expect(logout.mock.calls.length).toEqual(1);
      expect(logout).toBeCalled();
    });
  });

  describe('redux', () => {
    it('mapStateToProps', () => {
      const fakeState = { locale: { translations } };
      const props = mapStateToProps(fakeState);
      expect(props).toEqual({
        translations: {
          home__loggedin: translations.home__loggedin,
          home__welcome: translations.home__welcome,
          logout: translations.logout,
        },
      });
    });

    it('mapDispatchToProps', () => {
      const mockDispatch = jest.fn();
      const props = mapDispatchToProps(mockDispatch);
      expect(Object.keys(props)).toEqual(['logout']);
      expect(typeof props.logout).toEqual('function');
      props.logout();

      expect(mockDispatch.mock.calls.length).toEqual(1);
      expect(mockDispatch).toBeCalledWith(logoutAction());
    });
  });
});
