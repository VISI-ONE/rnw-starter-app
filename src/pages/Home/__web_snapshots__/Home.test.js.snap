// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page rendering render properly - web 1`] = `
<div
  className="rn-alignItems-1awozwy rn-alignSelf-1pz39u2 rn-backgroundColor-1tvv03i rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flex-13awgt0 rn-flexGrow-1m1wadx rn-flexShrink-1awmn5t rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-paddingTop-m611by rn-position-bnwqim rn-zIndex-1lgpqti"
>
  <div
    className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flex-13awgt0 rn-flexGrow-1m1wadx rn-flexShrink-1awmn5t rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <div
      className="rn-alignSelf-1kihuf0 rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1inkyih rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
    >
      Welcome to RNW starter app
    </div>
    <div
      className="rn-alignItems-1oszu61 rn-backgroundColor-wib322 rn-backgroundPosition-vvn4in rn-backgroundRepeat-u6sd8q rn-backgroundSize-4gszlv rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-height-erd55g rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-width-1ipicw7 rn-zIndex-1lgpqti"
    />
    <div
      className="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-18u37iz rn-justifyContent-1777fci rn-marginTop-1g94qm0 rn-marginRight-7o8qx1 rn-marginBottom-d0pm55 rn-marginLeft-1f6r7vd rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
    >
      <div
        className="rn-alignItems-1oszu61 rn-backgroundColor-wib322 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-cursor-1loqt21 rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1g94qm0 rn-marginRight-7o8qx1 rn-marginBottom-d0pm55 rn-marginLeft-1f6r7vd rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-touchAction-1gvxusu rn-userSelect-lrvibr rn-zIndex-1lgpqti"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        tabIndex="0"
      >
        <div
          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-759hku rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
          dir="auto"
        >
          Logout
        </div>
      </div>
    </div>
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
    >
      You are now logged in
    </div>
  </div>
</div>
`;

exports[`Home page rendering render properly with qr code - web 1`] = `
<div
  className="rn-alignItems-1awozwy rn-alignSelf-1pz39u2 rn-backgroundColor-1tvv03i rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flex-13awgt0 rn-flexGrow-1m1wadx rn-flexShrink-1awmn5t rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-paddingTop-m611by rn-position-bnwqim rn-zIndex-1lgpqti"
>
  <div
    className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flex-13awgt0 rn-flexGrow-1m1wadx rn-flexShrink-1awmn5t rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <div
      className="rn-alignSelf-1kihuf0 rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1inkyih rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
    >
      Welcome to RNW starter app
    </div>
    <div
      className="rn-alignItems-1oszu61 rn-backgroundColor-wib322 rn-backgroundPosition-vvn4in rn-backgroundRepeat-u6sd8q rn-backgroundSize-4gszlv rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-height-erd55g rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-width-1ipicw7 rn-zIndex-1lgpqti"
    />
    <div
      className="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-18u37iz rn-justifyContent-1777fci rn-marginTop-1g94qm0 rn-marginRight-7o8qx1 rn-marginBottom-d0pm55 rn-marginLeft-1f6r7vd rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
    >
      <div
        className="rn-alignItems-1oszu61 rn-backgroundColor-wib322 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-cursor-1loqt21 rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1g94qm0 rn-marginRight-7o8qx1 rn-marginBottom-d0pm55 rn-marginLeft-1f6r7vd rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-touchAction-1gvxusu rn-userSelect-lrvibr rn-zIndex-1lgpqti"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        tabIndex="0"
      >
        <div
          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-759hku rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
          dir="auto"
        >
          Logout
        </div>
      </div>
    </div>
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
    >
      You are now logged in
    </div>
  </div>
</div>
`;
